x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentNodeService.java

```diff
@@ -238,16 +241,33 @@ void loadQueuedModels() {
     }
 
     public void stopDeploymentAndNotify(TrainedModelDeploymentTask task, String reason, ActionListener<AcknowledgedResponse> listener) {
+        Optional<RoutingInfo> routingInfo = getRoutingInfo(task.getModelId());
+        if (routingInfo.isEmpty()) {
+            return;
+        }
```

Comment:
I am not sure about this. This just indicates that the cluster state doesn't have this route. It may be that the task is still loaded and model is around. Do we know for sure that exiting here won't leave a zombie process?

Reply:
Good point. I was also not notifying the listener. I'll add a call to `deploymentManager.stopDeployment` and notify the listener. This should ensure the model is cleaned up in this edge case.